nautobot/extras/views.py

```diff
@@ -967,13 +967,29 @@ def _get_job_model_or_404(self, class_path=None, slug=None):
     def get(self, request, class_path=None, slug=None):
         job_model = self._get_job_model_or_404(class_path, slug)
 
+        initial = normalize_querydict(request.GET)
+        if "kwargs_from_job_result" in initial:
+            job_result_pk = initial.pop("kwargs_from_job_result")
+            try:
+                job_result = JobResult.objects.get(pk=job_result_pk)
+                # Allow explicitly specified arg values in request.GET to take precedence over the saved job_kwargs,
+                # for example "?kwargs_from_job_result=<UUID>&integervar=22&_commit=False"
+                explicit_initial = initial
+                initial = job_result.job_kwargs.get("data", {}).copy()
+                commit = job_result.job_kwargs.get("commit")
```

Comment:
Let's do a cursory check that the source Job's are of the same `job_class`. One thing that we can run into is if the previous Job had kwargs of the same name but different type you might get unexpected or undesirable outcomes. This would be hard unless we pickle with types too. At the very least we can minimize sniffing the values of a Job I am not allowed to run/see by knowing/guessing/finding a past-run's JobResultID and seeing the values on a Job I do have access to. Maybe we just check it's the same Job Model? Both?

Reply:
Could be as simple as:
```suggestion
                job_result = job_model.results.get(pk=job_result_pk)
                # Allow explicitly specified arg values in request.GET to take precedence over the saved job_kwargs,
                # for example "?kwargs_from_job_result=<UUID>&integervar=22&_commit=False"
                explicit_initial = initial
                initial = job_result.job_kwargs.get("data", {}).copy()
                commit = job_result.job_kwargs.get("commit")
```